core/src/main/java/org/apache/iceberg/BaseFilesTable.java

```diff
@@ -177,14 +177,17 @@ public CloseableIterable<StructLike> rows() {
         Set<Integer> readableMetricsIds = TypeUtil.getProjectedIds(readableMetricsField.type());
         Schema fileProjection = TypeUtil.selectNot(projection, readableMetricsIds);
 
+        int position = projection.columns().indexOf(readableMetricsField);
```

Comment:
could it be possible that the projection schema doesn't contain readable metrics struct?

Reply:
I guess this is the else case (where the field exists in user-projection).

Reply:
ah. didn't see the if-else before this

Reply:
nit: maybe remove the empty line before this.